src/core/getters/enum.ts

```diff
@@ -20,11 +20,32 @@ export const getEnumImplementation = (value: string, type: string) => {
     // nullable value shouldn't be in the enum implementation
     if (val === 'null') return acc;
     const isTypeNumber = type === 'number';
-    const isNumber = !Number.isNaN(Number(val.slice(1, -1)));
-    const key =
-      isNumber || isTypeNumber
-        ? toNumberKey(isTypeNumber ? val.toString() : val.slice(1, -1))
-        : sanitize(val, { underscore: '_', whitespace: '_', dash: '-' });
+
+    const valueWithoutQuotes = isTypeNumber ? val.toString() : val.slice(1, -1);
+    const isNumber = isTypeNumber || !Number.isNaN(Number(valueWithoutQuotes));
+
+    let key: string;
+    if (isNumber) {
+      key = toNumberKey(valueWithoutQuotes);
+    } else {
+      key = sanitize(valueWithoutQuotes, {
+        underscore: '_',
+        whitespace: '_',
+        dash: '-',
+      });
+
+      /*
+        Potentially sanitize can strip to much from a value
+        For example:
+        sanitize("привет, мир!") -> "_"
+        sanitize("❤") -> ""
+        In that case it's more reasonable to return a full unchanged value
+        which will be later surrounded with quotes
+       */
+      if (key.length < valueWithoutQuotes.length) {
+        key = valueWithoutQuotes;
```

Comment:
You could have case where you don't replace the space by _ anymore with this which was a convention added from the beginning

Reply:
Sorry, I'm not sure what you mean

Unicode symbols in enum values are trimmed in `sanitize` function with the regex below
https://github.com/anymaniax/orval/blob/3f7a468afa0eb7db60000a2ce251eda996366905/src/utils/string.ts#L66

As a result some keys are reduced to empty strings. For example
```ts
sanitize('привет') // ''
```

Maybe you could suggest a better idea how to preserve original unicode values?

Reply:
With your implementation, if you have something like this `привет  привет` the space is not replaced by an underscore and that's the issue because by convention since the beginning orval does that and so it's a breaking change

Reply:
Thanks, for the explanation. I've fixed it 🙏